did/peer.go

```diff
@@ -0,0 +1,539 @@
+//                    DID Peer
+// ------------------------------------------------
+// https://identity.foundation/peer-did-method-spec/
+//
+// Peer based, self signed DID method.
+//
+// The method can be used independent of any central source of truth, and is
+// intended to be cheap, fast, scalable, and secure. It is suitable for most
+// private relationships between people, organizations, and things. We expect
+// that peer-to-peer relationships in every blockchain ecosystem can benefit by
+// offloading pairwise and n-wise relationships to peer DIDs.
+//
+// Currently only methods 0 and 2 are supported.
+// Method 1 will be supported in a future date
+package did
+
+import (
+	gocrypto "crypto"
+	b64 "encoding/base64"
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/TBD54566975/ssi-sdk/crypto"
+	"github.com/TBD54566975/ssi-sdk/util"
+	"github.com/pkg/errors"
+)
+
+type DIDPeer string
+type PurposeType string
+
+var availablePeerMethods = map[string]Resolver{
+	"0": PeerMethod0{},
+	"1": PeerMethod1{},
+	"2": PeerMethod2{},
+}
+
+// ANBF specified here:
+// https://identity.foundation/peer-did-method-spec/#method-specific-identifier
+const (
+	PeerMethodPrefix                = "peer"
+	DIDPrefix                       = "did"
+	PeerTransform                   = "z"
+	PeerEncNumBasis                 = Base58BTCMultiBase
+	PeerDIDRegex                    = `^did:peer:(([01](z)([1-9a-km-zA-HJ-NP-Z]{46,47}))|(2((\.[AEVID](z)([1-9a-km-zA-HJ-NP-Z]{46,47}))+(\.(S)[0-9a-zA-Z=]*)?)))$`
+	PeerKnownContext                = "https://w3id.org/did/v1"
```

Comment:
this is a duplicate of https://github.com/TBD54566975/ssi-sdk/blob/main/did/model.go#L12